src/reps/array.js

```diff
@@ -19,184 +19,114 @@ const DOM = React.DOM;
  * Renders an array. The array is enclosed by left and right bracket
  * and the max number of rendered items depends on the current mode.
  */
-let ArrayRep = React.createClass({
-  displayName: "ArrayRep",
```

Comment:
So I tested it with my test case of 1000 reps (launched 10 times to get an average) :  
React.createClass : ~ 3500ms
Functions + React.createElement: ~2200ms 
Functions : ~1600ms

Function is definitely faster and I don't think the benefit we get from doing React.createElement (having propTypes warnings) justify the loss of performance.